File: src/main/java/com/anfema/ionclient/authentication/BasicAuth.java
package com.anfema.ionclient.authentication;

import android.util.Base64;

import com.anfema.ionclient.utils.IonLog;

import java.io.UnsupportedEncodingException;

/**
 * Helper class for basic access authentication.
 */
public class BasicAuth
{
	/**
	 * Generates the value for the "Authorization" header.
	 */
	public static String getAuthHeaderValue( String username, String password )
	{
		String usernamePassword = username + ":" + password;
		byte[] data = null;
		try
		{
			data = usernamePassword.getBytes( "UTF-8" );
		}
		catch ( UnsupportedEncodingException e )
		{
			IonLog.ex( new Exception( "Could not get bytes of " + usernamePassword, e ) );
		}
		String base64 = Base64.encodeToString( data, Base64.NO_WRAP );
		return "Basic " + base64;
	}
}

File: src/main/java/com/anfema/ionclient/authentication/BasicAuth.kt
package com.anfema.ionclient.authentication

import okhttp3.Credentials

/**
 * Helper class for basic access authentication.
 */
object BasicAuth {
    /**
     * Generates the value for the "Authorization" header.
     */
    @JvmStatic
    @Deprecated("Use implementation from OkHttp",
        ReplaceWith("Credentials.basic(username, password)", "okhttp3.Credentials"))
    fun getAuthHeaderValue(username: String, password: String): String =
        Credentials.basic(username, password)
}
